sympy/functions/special/gamma_functions.py

```diff
@@ -1050,7 +1227,184 @@ def trigamma(x):
     .. [2] http://mathworld.wolfram.com/TrigammaFunction.html
     .. [3] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
-    return polygamma(1, x)
+    def _eval_evalf(self, prec):
+        n = sympify(1)
+        # the mpmath polygamma implementation valid only for nonnegative integers
+        if n.is_number and n.is_real:
+            if (n.is_integer or n == int(n)) and n.is_nonnegative:
+                return super(trigamma, self)._eval_evalf(prec)
+
+    def fdiff(self, argindex=1):
+        if argindex == 1:
+            n = sympify(1)
+            z = self.args[0]
+            return polygamma(n + 1, z)
+        else:
+            raise ArgumentIndexError(self, argindex)
```

Comment:
Probably, filtering incorrect inputs before beginning the processing will be better,
```python
if argindex != 1:
            raise ArgumentIndexError(self, argindex)
n = sympify(1)
z = self.args[0]
return polygamma(n + 1, z)
```